research/delf/delf/python/training/global_features/train.py

```diff
@@ -0,0 +1,719 @@
+                     help='Update model weights every N batches, used to '
+                          'handle relatively large batches, ' +
+                          'batch_size effectively becomes update_every x '
+                          'batch_size (default: 1).')
+flags.DEFINE_enum('optimizer', default='adam', enum_values=_OPTIMIZER_NAMES,
+                  help='Optimizer options: ' +
+                       ' | '.join(_OPTIMIZER_NAMES) +
+                       ' (default: adam).')
+flags.DEFINE_float('lr', default=1e-6,
+                   help='Initial learning rate (default: 1e-6).')
+flags.DEFINE_float('momentum', default=0.9, help='Momentum.')
+flags.DEFINE_float('weight_decay', default=1e-6,
+                   help='Weight decay (default: 1e-6).')
+flags.DEFINE_integer('print_freq', default=10,
+                     help='Print frequency (default: 10).')
+flags.DEFINE_bool('resume', default=False,
+                  help='Whether to start from the latest checkpoint in the '
+                       'logdir.')
+flags.DEFINE_bool('launch_tensorboard', False,
+                  help='Whether to launch tensorboard.')
+
+
+def main(argv):
+  # Manually check if there are unknown test datasets and if the dataset
+  # ground truth files are downloaded.
+  for dataset in FLAGS.test_datasets.split(','):
+    if dataset not in _TEST_DATASET_NAMES:
+      raise ValueError('Unsupported or unknown test dataset: {}.'.format(
+              dataset))
+
+    data_dir = os.path.join(FLAGS.data_root, 'gnd_{}.pkl'.format(dataset))
+    if not os.path.isfile(data_dir):
+      raise ValueError(
+              '{} ground truth file at {} not found. Please download it '
+              'according to '
+              'the DELG instructions.'.format(dataset, FLAGS.data_root))
+
+  # Check if train dataset is downloaded and dowload it if not found.
+  sfm120k.download_train(FLAGS.data_root)
+
+  # Creating model export directory if it does not exist.
+  model_directory = global_features_utils.create_model_directory(
+          FLAGS.training_dataset, FLAGS.arch, FLAGS.pool, FLAGS.whitening,
+          FLAGS.pretrained, FLAGS.loss, FLAGS.loss_margin, FLAGS.optimizer,
+          FLAGS.lr,
+          FLAGS.weight_decay, FLAGS.neg_num, FLAGS.query_size, FLAGS.pool_size,
+          FLAGS.batch_size, FLAGS.update_every, FLAGS.image_size,
+          FLAGS.directory)
+
+  # Setting up logging directory, same as where the model is stored.
+  logging.get_absl_handler().use_absl_log_file('absl_logging', model_directory)
+
+  # Set cuda visible device.
+  os.environ['CUDA_VISIBLE_DEVICES'] = FLAGS.gpu_ids
+  global_features_utils.debug_and_log('>> Num GPUs Available: {}'.format(
+          len(tf.config.experimental.list_physical_devices('GPU'))),
+          FLAGS.debug)
+
+  # Allow memory growth.
+  gpus = tf.config.experimental.list_physical_devices('GPU')
+  if gpus:
+    try:
+      for gpu in gpus:
+        tf.config.experimental.set_memory_growth(gpu, True)
```

Comment:
It allows to see how much GPU memory is actually occupied by the code because under default conditions tensorflow takes up all of the available GPU memory.

Reply:
Got it, thanks for explaining! I saw you removed it, but feel free to keep in case you think it's helpful :)